### PR TITLE
Check imported translations for warnings

### LINKED ISSUE #401: Imported translations should be checked for warnings
When you add translations via file import, formatting errors such as non-matching placeholders/tags are left undetected. Errors should be flagged and listed under the Warnings tab.

Previously: https://glotpress.trac.wordpress.org/ticket/396

### PR DESCRIPTION
**Pull request by yoavf**
Fixes #401

### CODE REVIEW

**Comment by yoavf** [comment]
This still imports the translation as active (or fuzzy, depending on filter). Should we set those to waiting by default?

**Comment by ocean90** [comment]
IMO when they are set to waiting the chance will be higher that they will be reviewed, warnings are usually ignored...

**Comment by yoavf** [comment]
Agreed - what we've done so far is to add `$entry` to the `gp_translation_set_import_status` filter, and then use a plugin to change the status to waiting. 
I think this should be the default though, and allow an overwrite via the filter.

If that sounds good, I'll update the PR for that

**Comment by ocean90** [comment]
Sounds good to me. It's a bit related to https://github.com/GlotPress/GlotPress-WP/pull/378/files#diff-131463cb25fdae42d7139e8116843697R174

**Comment by yoavf** [comment]
Now setting to waiting by default is has warnings.

File: gp-includes/things/translation-set.php [hunk 1]
```diff
@@ -192,17 +192,25 @@ public function import( $translations ) {
 					$entry->user_id = $user->ID;
 				}
 
+				$entry->status = $is_fuzzy ? 'fuzzy' : 'current';
```
Comment (toolstack) [inline gp-includes/things/translation-set.php:195]: More readable to use a full  if() {} .

File: gp-includes/things/translation-set.php [hunk 2]
```diff
@@ -192,17 +192,25 @@ public function import( $translations ) {
...
 				/**
 				 * Filter the the status of imported translations of a translation set.
 				 *
 				 * @since 1.0.0
 				 *
-				 * @param string $status The status of imported translations.
+				 * @param string            $status The status of imported translations.
+				 * @param Translation_Entry $entry  Translation entry object to import.
 				 */
-				$entry->status = apply_filters( 'gp_translation_set_import_status', $is_fuzzy ? 'fuzzy' : 'current' );
```
Comment (toolstack) [inline gp-includes/things/translation-set.php:204]: More readable to use a full  if() {} .

**Pull request merged**